docs/frontend-system/building-apps/08-migrating.md

```diff
@@ -0,0 +1,439 @@
+
+```ts
+const app = createApp({
+  apis: [
+    createApiFactory({
+      api: scmIntegrationsApiRef,
+      deps: { configApi: configApiRef },
+      factory: ({ configApi }) => ScmIntegrationsApi.fromConfig(configApi),
+    }),
+  ],
+});
+```
+
+Can be converted to the following extension:
+
+```ts
+const scmIntegrationsApi = createApiExtension({
+  factory: createApiFactory({
+    api: scmIntegrationsApiRef,
+    deps: { configApi: configApiRef },
+    factory: ({ configApi }) => ScmIntegrationsApi.fromConfig(configApi),
+  }),
+});
+```
+
+### `icons`
+
+Icons are currently installed through the usual options to `createApp`, but will be switched to use extensions in the future.
+
+### `plugins`
+
+Plugins are now passed through the `features` options instead.
+
+### `featureFlags`
+
+Declaring features flags in the app is no longer supported, move these declarations to the appropriate plugins instead.
+
+### `components`
+
+Many app components are now installed as extensions instead using `createComponentExtension`. See the section on [configuring app components](./index.md#TODO) for more information.
+
+The `Router` component is now a built-in extension that you can override using `createRouterExtension`.
+
+The Sign-in page is now installed as an extension using the `createSignInPageExtension` instead. See the section on [configuring the sign-in page](./index.md#TODO) for more information.
+
+For example, the following sign-in page configuration:
+
+```tsx
+const app = createApp({
+  components: {
+    SignInPage: props => (
+      <SignInPage
+        {...props}
+        provider={{
+          id: 'github-auth-provider',
+          title: 'GitHub',
+          message: 'Sign in using GitHub',
+          apiRef: githubAuthApiRef,
+        }}
+      />
+    ),
+  },
+});
+```
+
+Can be converted to the following extension:
+
+```tsx
+const signInPage = createSignInPageExtension({
+  loader: async () => props =>
+    (
+      <SignInPage
+        {...props}
+        provider={{
+          id: 'github-auth-provider',
+          title: 'GitHub',
+          message: 'Sign in using GitHub',
+          apiRef: githubAuthApiRef,
+        }}
+      />
+    ),
+});
+```
+
+### `themes`
+
+Themes are now installed as extensions instead using `createThemeExtension`. See the section on [configuring themes](./index.md#TODO) for more information.
+
+For example, the following theme configuration:
+
+```tsx
+const app = createApp({
+  themes: [
+    {
+      id: 'light',
+      title: 'Light',
+      variant: 'light',
+      Provider: ({ children }) => (
+        <UnifiedThemeProvider theme={customLightTheme} children={children} />
+      ),
+    },
+  ],
+```
+
+Can be converted to the following extension:
+
+```tsx
+const lightTheme = createThemeExtension({
+  id: 'light',
+  title: 'Light Theme',
+  variant: 'light',
+  icon: <LightIcon />,
+  Provider: ({ children }) => (
+    <UnifiedThemeProvider theme={builtinThemes.light} children={children} />
```

Comment:
```suggestion
    <UnifiedThemeProvider theme={builtinThemes.light}>
      {children}
    </UnifiedThemeProvider>
```

it's kinda a bit clearer and i also see some lint rules and best practices avoiding a `children` prop. 🤷